### PR TITLE
[REM] crm, *: remove team_id from res.partner model

### PR DESCRIPTION
**Pull request by adsh-odoo**
*= sale, sales_team, website_crm_partner_assign, website_sale

**Purpose**
Clean up the Sales Team/Partner relationship following some feedback that proves
it's currently a little messy.

**Specifications**
Remove team_id field from the res.partner model because we mainly use the user_id
field instead. Having both fields causes confusion, especially regarding which team
the salesperson belongs to.

**Test cases were changed/removed accordingly.**

Upgrade PR: https://github.com/odoo/upgrade/pull/5609

Task-3618859

### CODE REVIEW

**Comment by fpodoo** [comment]
What's the point of removing from the view, but keeping the field? I believe we should keep the field, but make a computed stored field, like on sales order or invoices.

**Comment by thsh-odoo** [comment]
Hello @tde-banana-odoo 

Changes are done.
Can you please have a look?

Thanks :slightly_smiling_face:

File: addons/website_crm_partner_assign/report/crm_partner_report.py [hunk 1]
```diff
@@ -26,7 +25,7 @@ class CrmPartnerReportAssign(models.Model):
         'account.invoice.report': ['invoice_date', 'partner_id', 'price_subtotal', 'state', 'move_type'],
         'crm.lead': ['partner_assigned_id'],
         'res.partner': ['activation', 'country_id', 'date_partnership', 'date_review',
-                        'grade_id', 'parent_id', 'team_id', 'user_id'],
```
Comment (awa-odoo) [inline addons/website_crm_partner_assign/report/crm_partner_report.py:29]: Looks like we missed this one: 

https://github.com/odoo/odoo/blob/master/addons/website_crm_partner_assign/models/crm_lead.py#L85

?

**Review by awa-odoo — COMMENTED** [review]
Hello @thsh-odoo @adsh-odoo 

One comment about a possible missing occurrence, otherwise not much to say.

About the commit message, it mentions:
"1) Remove the team_id field from the res.partner model"
And then:
"2) Remove the field from the views (and there seem to be quite a few messy ones)"

But obviously if you removed the field completely you also have to remove it from the views, can you reword that part? It seems a little bit too much copy/pasted based on the task pad content.

Thanks for your work & cheers!

File: addons/website_crm_partner_assign/report/crm_partner_report.py (hunk 1 above)
Comment (thsh-odoo) [reply addons/website_crm_partner_assign/report/crm_partner_report.py:29]: Hello @awa-odoo 

changes are done.
commit msg is also changed :sweat_smile: .
Can you please have a look?

Thanks :slightly_smiling_face:

**Comment by awa-odoo** [comment]
@robodoo r+

**Comment by kharkot** [comment]
> What's the point of removing from the view, but keeping the field? I believe we should remove it completely from the object - or transform it on a related field based on the sales person.

@fpodoo Why did you suggest (and in the end decide) to remove that field? Several Odoo users have data in that field (for reporting purposes); it cannot be taken from a salesperson because a salesperson (sine R15) can be in several sales teams.
Such changes cause the removal of important data while upgrading.

**Pull request closed without merging**